backend/src/controllers/productsController.js

```diff
@@ -5,27 +5,50 @@ import path from "path";
 
 const uniqueId = uuidv4();
 
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = path.dirname(__filename);
-const filePath = path.join(__dirname, "../../../frontend/views/products.html");
+// const __filename = fileURLToPath(import.meta.url);
+// const __dirname = path.dirname(__filename);
+// const filePath = path.join(__dirname, "../../../frontend/views/products.html");
+// const detailsPath = path.join(
+//   __dirname,
+//   "../../../frontend/views/product-details-page.html"
+// );
 
 const getProducts = (req, res) => {
-  res.sendFile(filePath);
+  res.json(products);
 };
 
+// const getProductDetails = (req, res) => {
+//   res.sendFile(detailsPath);
+// };
+
 const addProduct = (req, res) => {
   const { productName, productDescription, price, stock } = req.body;
+  const requiredFields = [
+    "productName",
+    "productDescription",
+    "price",
+    "stock",
+  ];
+
+  const missingFields = requiredFields.filter((field) => !req.body[field]);
 
-  const newProduct = {
-    id: uniqueId,
-    productName: productName,
-    description: productDescription,
-    price: price,
-    stock: Number(stock),
-    time: new Date(),
-  };
-  products.push(newProduct);
-  res.json({ message: "product added successfully" });
+  if (missingFields.length > 0) {
+    res.status(422).json({
+      success: false,
+      message: "All fields are required",
+    });
+  } else {
+    const newProduct = {
+      id: uniqueId,
+      productName: productName,
+      description: productDescription,
+      price: price,
+      stock: Number(stock),
+      time: new Date(),
+    };
+    products.push(newProduct);
+    res.status(200).json({ message: "product added successfully" });
```

Comment:
also instead of just message, also add a field that returns the product object. It will be useful for the client app